src/components/customMdx/tabbedContent/index.tsx

```diff
@@ -1,17 +1,42 @@
 import * as React from 'react'
 import styled from 'styled-components'
 
+import { defaultOSIndex } from "./DefaultTabOS"
+
+enum DefaultTabType {
+  OS = 'OS',
+}
+
 interface CodeProps {
   tabs?: any[]
+  defaultTabType?: DefaultTabType
 }
 
 type CodeBlockProps = CodeProps & React.ReactNode
 
-const TabbedContent = ({ tabs, children }: CodeBlockProps) => {
+const TabbedContent = ({ tabs, defaultTabType, children }: CodeBlockProps) => {
   const [activeIndex, setActiveIndex] = React.useState(0)
   const tabContent =
     children &&
     children.filter((child: any) => child.props && child.props.originalType === 'tab')[activeIndex]
+
+  React.useEffect(() => {
+
+    if (tabs && tabs.length > 1) {
+      // check if any default tab behaviour has been stipulated
+      switch (defaultTabType) {
```

Comment:
Curious why you have used a switch here? 😊

Reply:
@molebox Great question! Although the switch is not really needed for a single "case", my thinking was that going forward we can use this structure to add different/additional override logic for a different set of tabs.

For example, if we have tabbed content for our _supported languages_ or _supported databases_ and wanted a "funnelled" journey where we set a default based on a users previous selection, we could then use this switch case to check which logic we would like to apply in determining the default value for a set of tabs.

Reply:
Awesome, thanks for explaining. 😊